### PR TITLE
Initial drupal configuration

### PR DESCRIPTION
**Pull request by thomscode**
Initial release to contain a basic Drupal installation. Version and dependency management handled by [composer](https://getcomposer.org/) and environment handled by [Docksal](https://docksal.io) using `docker-compose`.

### CODE REVIEW

**Comment by dmlb2000** [comment]
@thomscode Please put in an issue describing what you've done here 8500+ lines of code is a lot to ingest. I want to know where you got it, what did you run and in what order?

**Comment by dmlb2000** [comment]
trying to trigger travis

**Comment by thomscode** [comment]
### Project setup steps
1. Install Drupal with Composer
    - `composer create-project drupal-composer/drupal-project:8.x-dev pacifica-search --stability dev --no-interaction`
    - Instructions provided by [Drupal's example GitHub project](https://github.com/drupal-composer/drupal-project)
2. Setup appropriate [Docksal](https://docksal.io) files to manage a containerized development environment (uses docker-compose behind the scenes, but gives the added flexibility of having consistent setup and deployment scripts).
    - `fin vm start`
    - `cd <pacifica project folder>`
    - `fin init`

### File explanations and sources
* Files generated by/for Docksal
    - `.docksal/commands/init` - From [Docksal's Drupal 8 Example](https://github.com/docksal/drupal8/blob/master/.docksal/commands/init)
    - `.docksal/docksal.env` - Example file from [Docksal's github project](https://github.com/docksal/docksal/blob/develop/examples/.docksal/docksal.env)
    - `.docksal/docksal.yml` - Example file from [Docksal's github project](https://github.com/docksal/docksal/blob/develop/examples/.docksal/docksal.yml)
    - `.docksal/example.docksal-local.env` - Example of how to turn on XDebug - Manually created
* Files generated by composer project (all files either generated directly by `composer` or by Drupal's [composer project](https://github.com/drupal-composer/drupal-project))
    - `.gitignore` - Default Git ignore file
    - `.travis.yml` - Default TravisCI file
    - `LICENSE` - Should likely either be deleted or updated based on Pacifica's license
    - `composer.json` - base project composer file
    - `composer.lock` - Composer version lock file
    - `config/sync/.htaccess` - Ensures directory security in Apache
    - `config/sync/README.txt` - Explains the purpose of this directory
    - `docroot/.csslintrc` - Allows for CSS Linting and rules to be enforced
    - `docroot/.editorconfig` - Drupal [editor config](http://editorconfig.org/) file
    - `docroot/.eslintignore` - [ESLint](http://eslint.org/) configuration
    - `docroot/.eslintrc.json` - [ESLint](http://eslint.org/) configuration
    - `docroot/.gitattributes` - Drupal Git configuration
    - `docroot/.htaccess` - Default Drupal file to manage paths in Apache properly
    - `docroot/autoload.php` - Composer generated PHP class autoloader
    - `docroot/index.php` - Main Drupal index file
    - `docroot/modules/.gitkeep` - Empty hidden file to keep proper directory structure in the repository
    - `docroot/profiles/.gitkeep` - Empty hidden file to keep proper directory structure in the repository
    - `docroot/robots.txt` - Default Drupal file
    - `docroot/sites/default/default.services.yml` - Default services to setup and enable in production environment
    - `docroot/sites/default/default.settings.local.php` - Default local development Drupal settings file (used to ensure settings are properly setup in local development environment)
    - `docroot/sites/default/default.settings.php` - Default Drupal settings file (used to ensure settings exist in local development environment)
    - `docroot/sites/development.services.yml` - File to hold service declarations for use in development (testing and other things you don't want enabled in production)
    - `docroot/sites/example.settings.local.php` - Example local development settings file that includes appropriate default docksal configuration
    - `docroot/sites/example.sites.php` - Example file to define a multi-site Drupal instance. Can probably be removed but may be useful if anyone wants to create a Pacifica multi-site Drupal installation
    - `docroot/themes/.gitkeep` - Empty hidden file to keep proper directory structure in the repository
    - `docroot/update.php` - Drupal PHP file to handle database updates
    - `docroot/web.config` - I believe this has something to do with Apache, but I don't know enough about Apache to know for sure
    - `drush/README.md` - Explanation of directory purpose
    - `drush/policy.drush.inc` - Ensure Drupal module management is handled by `composer` instead of `drush`
    - `phpunit.xml.dist` - I don't know enough about PHP Unit to know what this does. I'm assuming it is a default configuration for PHP Unit tests
    - `scripts/composer/ScriptHandler.php` - Default scripts to ensure proper file structure is created for Drupal to run properly


**Note:** Changes to individual files are marked on the file's themselves

**Review by thomscode — COMMENTED** [review]
This file is unchanged from source location

File: .docksal/commands/init [hunk 1]
```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env bash
```
Comment (thomscode) [inline .docksal/commands/init:1]: This file is unchanged from source location

File: .docksal/docksal.env [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# This is a shared configuration file that is intended to be stored in the project repo.
+# To override a variable locally:
+# - create .docksal/docksal-local.env file and local variable overrides there
+# - add .docksal/docksal-local.env to .gitignore
+
+# Use the default Docksal stack
+DOCKSAL_STACK=default
+
+# Docksal configuration.
+VIRTUAL_HOST=pacifica.docksal
```
Comment (thomscode) [inline .docksal/docksal.env:10]: This line modification and the removal of the containers to use are the only modifications to this file from it's source location

File: .docksal/docksal.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+version: "2.1"
+services:
+  web:
+    hostname: web
+    image: docksal/web:latest-apache2.4
```
Comment (thomscode) [inline .docksal/docksal.yml:5]: Changed Apache version from 2.2. to 2.4

File: .docksal/docksal.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      - APACHE_DOCUMENTROOT=/var/www/${DOCROOT:-docroot}
+    depends_on:
+      - cli
+    dns:
+      - ${DOCKSAL_DNS1}
+      - ${DOCKSAL_DNS2}
+
+  db:
+    hostname: db
+    image: docksal/db:mysql-5.7
```
Comment (thomscode) [inline .docksal/docksal.yml:21]: Changed MySQL version from 5.5. to 5.7

File: .docksal/docksal.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      - MYSQL_USER=${MYSQL_USER:-user}
+      - MYSQL_PASSWORD=${MYSQL_PASSWORD:-user}
+      - MYSQL_DATABASE=${MYSQL_DATABASE:-default}
+    dns:
+      - ${DOCKSAL_DNS1}
+      - ${DOCKSAL_DNS2}
+
+  cli:
+    hostname: cli
+    image: docksal/cli:php7
```
Comment (thomscode) [inline .docksal/docksal.yml:37]: Changed tag used for PHP 7 to use most recent image

File: .docksal/example.docksal-local.env [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# Enable/disable xdebug
+# To override locally, copy the two lines below into .docksal/docksal-local.env and adjust as necessary
+XDEBUG_ENABLED=1
```
Comment (thomscode) [inline .docksal/example.docksal-local.env:3]: This file is manually created but code is copied directly from `docksal.env` with only the value of the `XDEBUG_ENABLED` variable being changed

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,21 @@
+# Ignore directories generated by Composer
```
Comment (thomscode) [inline .gitignore:1]: This file is unchanged from source location

File: .travis.yml [hunk 8]
```diff
@@ -0,0 +1,40 @@
+language: php
```
Comment (thomscode) [inline .travis.yml:1]: This file is unchanged from source location

File: LICENSE [hunk 9]
```diff
@@ -0,0 +1,339 @@
+                    GNU GENERAL PUBLIC LICENSE
```
Comment (thomscode) [inline LICENSE:1]: This file is unchanged from source location

File: composer.json [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+            "DrupalProject\\composer\\ScriptHandler::createRequiredFiles"
+        ]
+    },
+    "extra": {
+        "installer-paths": {
+            "docroot/core": ["type:drupal-core"],
+            "docroot/libraries/{$name}": ["type:drupal-library"],
+            "docroot/modules/contrib/{$name}": ["type:drupal-module"],
+            "docroot/profiles/contrib/{$name}": ["type:drupal-profile"],
+            "docroot/themes/contrib/{$name}": ["type:drupal-theme"],
```
Comment (thomscode) [inline composer.json:71]: Lines 67-71 changed from `web/` to `docroot/`

File: composer.lock [hunk 11]
```diff
@@ -0,0 +1,5925 @@
+{
```
Comment (thomscode) [inline composer.lock:1]: Generated by composer with current drupal module and php package versions

File: config/sync/.htaccess [hunk 12]
```diff
@@ -0,0 +1,23 @@
+# Deny all requests from Apache 2.4+.
```
Comment (thomscode) [inline config/sync/.htaccess:1]: This file is unchanged from source location

File: config/sync/README.txt [hunk 13]
```diff
@@ -0,0 +1 @@
+This directory contains configuration to be imported into your Drupal site. To make this configuration active, visit admin/config/development/configuration/sync. For information about deploying configuration between servers, see https://www.drupal.org/documentation/administer/config
```
Comment (thomscode) [inline config/sync/README.txt:1]: This file is unchanged from source location

File: docroot/.csslintrc [hunk 14]
```diff
@@ -0,0 +1,40 @@
+--errors=box-model,
```
Comment (thomscode) [inline docroot/.csslintrc:1]: This file is unchanged from source location

File: docroot/.editorconfig [hunk 15]
```diff
@@ -0,0 +1,17 @@
+# Drupal editor configuration normalization
```
Comment (thomscode) [inline docroot/.editorconfig:1]: This file is unchanged from source location

File: docroot/.eslintignore [hunk 16]
```diff
@@ -0,0 +1,9 @@
+core/assets/vendor/**/*
```
Comment (thomscode) [inline docroot/.eslintignore:1]: This file is unchanged from source location

File: docroot/.eslintrc.json [hunk 17]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (thomscode) [inline docroot/.eslintrc.json:1]: This file is unchanged from source location

File: docroot/.gitattributes [hunk 18]
```diff
@@ -0,0 +1,56 @@
+# Drupal git normalization
```
Comment (thomscode) [inline docroot/.gitattributes:1]: This file is unchanged from source location

File: docroot/.htaccess [hunk 19]
```diff
@@ -0,0 +1,189 @@
+#
```
Comment (thomscode) [inline docroot/.htaccess:1]: This file is unchanged from source location

File: docroot/autoload.php [hunk 20]
```diff
@@ -0,0 +1,17 @@
+<?php
```
Comment (thomscode) [inline docroot/autoload.php:1]: Generated by composer

File: docroot/index.php [hunk 21]
```diff
@@ -0,0 +1,22 @@
+<?php
```
Comment (thomscode) [inline docroot/index.php:1]: This file is unchanged from source location

File: docroot/robots.txt [hunk 22]
```diff
@@ -0,0 +1,61 @@
+#
```
Comment (thomscode) [inline docroot/robots.txt:1]: This file is unchanged from source location

File: docroot/sites/default/default.services.yml [hunk 23]
```diff
@@ -0,0 +1,174 @@
+parameters:
```
Comment (thomscode) [inline docroot/sites/default/default.services.yml:1]: This file is unchanged from source location

File: docroot/sites/default/default.settings.local.php [hunk 24]
```diff
@@ -0,0 +1,128 @@
...
+ *
+ * The system module will periodically check the permissions of your site's
+ * site directory to ensure that it is not writable by the website user. For
+ * sites that are managed with a version control system, this can cause problems
+ * when files in that directory such as settings.php are updated, because the
+ * user pulling in the changes won't have permissions to modify files in the
+ * directory.
+ */
+$settings['skip_permissions_hardening'] = TRUE;
+
```
Comment (thomscode) [inline docroot/sites/default/default.settings.local.php:116]: Added all code below this comment based on [Docksal Drupal 8 example](https://github.com/docksal/drupal8/blob/master/docroot/sites/default/default.settings.local.php#L123)

The remainder of this file is auto generated by composer project and is unchanged.

File: docroot/sites/development.services.yml [hunk 25]
```diff
@@ -0,0 +1,9 @@
+# Local development services.
```
Comment (thomscode) [inline docroot/sites/development.services.yml:1]: This file is unchanged from source location

File: docroot/sites/example.settings.local.php [hunk 26]
```diff
@@ -0,0 +1,128 @@
+<?php
```
Comment (thomscode) [inline docroot/sites/example.settings.local.php:1]: Same as `default.settings.local.php` above and might be able to be removed.

File: docroot/sites/example.sites.php [hunk 27]
```diff
@@ -0,0 +1,55 @@
+<?php
```
Comment (thomscode) [inline docroot/sites/example.sites.php:1]: This file is unchanged from source location

File: docroot/update.php [hunk 28]
```diff
@@ -0,0 +1,31 @@
+<?php
```
Comment (thomscode) [inline docroot/update.php:1]: This file is unchanged from source location

File: docroot/web.config [hunk 29]
```diff
@@ -0,0 +1,103 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (thomscode) [inline docroot/web.config:1]: This file is unchanged from source location

File: drush/README.md [hunk 30]
```diff
@@ -0,0 +1 @@
+This directory contains commands, configuration and site aliases for Drush. See https://packagist.org/search/?type=drupal-drush for a directory of Drush commands installable via Composer.
```
Comment (thomscode) [inline drush/README.md:1]: This file is unchanged from source location

File: drush/policy.drush.inc [hunk 31]
```diff
@@ -0,0 +1,38 @@
+<?php
```
Comment (thomscode) [inline drush/policy.drush.inc:1]: This file is unchanged from source location

File: phpunit.xml.dist [hunk 32]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (thomscode) [inline phpunit.xml.dist:1]: This file is unchanged from source location

File: scripts/composer/ScriptHandler.php [hunk 33]
```diff
@@ -0,0 +1,100 @@
+<?php
```
Comment (thomscode) [inline scripts/composer/ScriptHandler.php:1]: This file is unchanged from source location

File: docroot/sites/default/default.settings.php [hunk 34]
```diff
@@ -0,0 +1,765 @@
...
+ * extensions.
+ *
+ * @see file_scan_directory()
+ * @see \Drupal\Core\Extension\ExtensionDiscovery::scanDirectory()
+ */
+$settings['file_scan_ignore_directories'] = [
+  'node_modules',
+  'bower_components',
+];
+
```
Comment (thomscode) [inline docroot/sites/default/default.settings.php:751]: Added the following import from [Docksal's Drupal 8 example](https://github.com/docksal/drupal8/blob/master/docroot/sites/default/default.settings.php#L742)

File: docroot/sites/default/default.settings.php [hunk 35]
```diff
@@ -0,0 +1,765 @@
+<?php
```
Comment (thomscode) [inline docroot/sites/default/default.settings.php:1]: This file is unchanged except for the changes commented at the end of the file.

File: .travis.yml (hunk 8 above)
Comment (thomscode) [reply .travis.yml:1]: Modified location of drupal installation: documented below

File: .travis.yml [hunk 36]
```diff
@@ -0,0 +1,40 @@
...
+
+script:
+  - if [[ $RELEASE = dev ]]; then composer --verbose remove --no-update drupal/console; fi;
+  - if [[ $RELEASE = dev ]]; then composer --verbose require --no-update drupal/core:8.4.x-dev drush/drush:9.0.x-dev; fi;
+  - if [[ $RELEASE = dev ]]; then composer --verbose update; fi;
+  - cd $TRAVIS_BUILD_DIR/docroot
+  - ./../vendor/bin/drush site-install --verbose --yes --db-url=sqlite://tmp/site.sqlite
+  - ./../vendor/bin/drush runserver $SIMPLETEST_BASE_URL &
+  - until curl -s $SIMPLETEST_BASE_URL; do true; done > /dev/null
+  # Skip core/tests/Drupal/Tests/ComposerIntegrationTest.php because docroot/ has no composer.json
```
Comment (thomscode) [inline .travis.yml:36]: Changed `web/` to `docroot/`

File: .travis.yml [hunk 37]
```diff
@@ -0,0 +1,40 @@
...
+  - composer --version
+
+install:
+  - composer --verbose install
+
+script:
+  - if [[ $RELEASE = dev ]]; then composer --verbose remove --no-update drupal/console; fi;
+  - if [[ $RELEASE = dev ]]; then composer --verbose require --no-update drupal/core:8.4.x-dev drush/drush:9.0.x-dev; fi;
+  - if [[ $RELEASE = dev ]]; then composer --verbose update; fi;
+  - cd $TRAVIS_BUILD_DIR/docroot
```
Comment (thomscode) [inline .travis.yml:32]: Changed `web/` to `docroot/`

**Comment by thomscode** [comment]
@dmlb2000, Travis seems to automatically trigger now. I have fixed a couple of things and am done commenting on the files, where they come from, and the changes I made to any of the files.

**Comment by thomscode** [comment]
Closing pull request as it doesn't conform to proper standards. Will be opening appropriate issues and creating new pull requests to resolve issues appropriately.

**Pull request closed without merging**